src/kernel/src/arch/aarch64/mod.rs

```diff
@@ -15,22 +16,88 @@ pub mod thread;
 mod start;
 
 pub use address::{VirtAddr, PhysAddr};
-pub use interrupt::send_ipi;
+pub use interrupt::{send_ipi, init_interrupts};
 pub use start::BootInfoSystemTable;
+use twizzler_abi::syscall::TimeSpan;
 
 pub fn init<B: BootInfo>(_boot_info: &B) {
     logln!("[arch::init] initializing exceptions");
     exception::init();
+
+    // intialize an instance of the timer
+    processor::enumerate_clocks();
+
+    use crate::time::TICK_SOURCES;
+    use twizzler_abi::syscall::{NanoSeconds, ClockSource, NANOS_PER_SEC, FEMTOS_PER_SEC};
+
+    let clk_src: u64 = ClockSource::BestMonotonic.into();
+    let cntp = &TICK_SOURCES.lock()[clk_src as usize];
+
+    let info = cntp.info();
+    logln!("[arch::timer] frequency: {} Hz, {} fs, {} ns", 
+        FEMTOS_PER_SEC / info.resolution().0, info.resolution().0, {
+            let femtos = info.resolution();
+            let nanos: NanoSeconds = femtos.into();
+            nanos.0
+        }
+    );
+
+    // read the timer
+    let t = cntp.read();
+    logln!("[arch::timer] current timer count: {}, uptime: {:?}",
+        t.value, t.value * t.rate
+    );
+
+    init_interrupts();
+
+    // set the timer interrupt to be routed to us
+    crate::machine::interrupt::INTERRUPT_CONTROLLER
+        .enable_interrupt(cntp::PhysicalTimer::INTERRUPT_ID);
+
+    // configure the timer to fire off interrupts
+    const WAIT_TIME: TimeSpan = TimeSpan::from_secs(3);
+
+    let phys_timer = cntp::PhysicalTimer::new();
+    phys_timer.set_timer(WAIT_TIME);
+
+    logln!("[arch::timer] firing off an interrupt in {} seconds...", WAIT_TIME.0.0);
```

Comment:
Shouldn't this be under a debug statement?

Reply:
I can wrap it under a `#[cfg(test)]` macro since this was only for testing purposes. In the future this section is going to be removed.